src/main/rx/spektrum.c

```diff
@@ -70,6 +70,11 @@ static bool spekHiRes = false;
 static bool srxlEnabled = false;
 static int32_t resolution;
 
+#ifdef USE_SPEKTRUM_RSSI_CH
```

Comment:
Using a define in here seems to be hackish. If this is a target specific thing, it should be done in the target definition.

Reply:
As far as I know, newbeedrone.com is going to release a new spektrum compatible receivers with this RSSI channel, then it's not only for this board.

Reply:
Why can't you just set `rssi_channel` in the CLI?

Reply:
@martinbudden It is similar to setting `rxConfigMutable()->rssi_channel = XXX` in the config.c. 
But the original driver will calculate the so-call RSSI using fade count, and put it into the buffer.

Reply:
Exactly, Use CLI or GUI to select any free rx channel for RSSI. 
Not sure how or if this can be done, but if you want it pre-defined as a default, it should be defined in some way in the target.h file.

Reply:
> As far as I know, newbeedrone.com is going to release a new spektrum compatible receivers with this RSSI channel, then it's not only for this board.

- if they are 'spektrum compatible' they should be built to be completely spektrum compatible, and not require special settings. Otherwise they are 'spektrum compatible but broken';
- it is not realistic to expect everybody who has bought one of these 'spektrum compatible' RXs to compile the firmware for whatever board they are using in order to be able to set the `USE_SPEKTRUM_RSSI_CH` compile time define.

Reply:
@mikeller 
I agree with you that a 'spektrum compatible' receiver should be completely matched with the official protocol. 
In terms of RSSI, the original driver we used for a long time calculates the RSSI from the first two bytes of the packet. Not sure if I get it wrong, these two bytes of a typical DSM2/DSMX packets are packet indexes and increased by like 1 every packet. Then the driver counts them and see how many packets were lost, so we know how is the link quality. That's different from RSSI (that's why it is called fake RSSI in the code) but still quite useful for flyings. 
It is not part of the protocol, so do the newbeedrone products.

Reply:
Is it acceptable to use `#undef USE_SPEKTRUM_FAKE_RSSI` and `#define USE_SPEKTRUM_RSSI_CH` in the target.h, or adding a variable like `uint8_t spektrum_use_rssi_ch` in the `rxConfig` and set it through CLI for flight controller using these devices.

Reply:
Why don't you just follow the advice given in https://github.com/betaflight/betaflight/pull/4775#discussion_r157448676 ?

The Spektrum RX code was updated to support proper, non-fake RSSI.

Reply:
- There is SRXL supported receiver with non-fake RSSI. 
- If I set `rssi_channel` in the CLI, and use spektrum 1024/2048 with `rssi_channel != 0`, the driver will get the RSSI from the first two bytes and put it into `spekChannelData[rssi_channel]`. 
- This PR is to find a way to use the original `spekChannelData[rssi_channel]` as the RSSI, no more calculation needed.

Reply:
Ok. Sounds like the facility to use non-fake RSSI should be extended to cover SRXL **and** 1024 / 2048, in a uniform way. @AndersHoglund, what is your take on this?

Reply:
The primary criterium is that a build flag should not be required to do this, since users should not have to re-flash their FC if they change their receiver. So it should be done on a config setting. Something line `rssi_using_fade` which can be set `on` or `off`.

Reply:
@martinbudden: I thought @AndersHoglund had already implemented something along these lines (for SRXL).

Reply:
The info I have from Spektrum says that only SRXL RXs sends real RSSI. This is why there is such a condition. 
It is really a mess in the received Rx frame, first and second byte can have three different meanings, depending on protocol in use:
1. SRXL: byte[0} 8-bit RSSI in dBm, in the -42 to -92 range. and -100 dBm if a frame is lost, due to fade, bad CRC etc.
2. Int bound DSM2/X: byte[0} 8-bit fade count, byte[1] System type.
3. Ext bound DSM2/X: byte[0] and byte[1] 16-bit BE fade count. 

There is no header info telling what the format is, except no 2 System Type. So that is why SRXL is used as an indirect condition for real RSSI. Maybe we could test for negative numbers in byte[0] instead. but that would clash with large (>128) fade counts with a type 2 Rx. An RSSI glitch like that in such a fade condition might be acceptable, it might even be a fail safe situation. Or having a new `use_spektrum_real_rssi` run time configuration flag, as @martinbudden suggests. But I would not like yet another manual config item (YAMCI). Not sure any of those changes lwould help this newbeedrone clone anyhow, if it is not fully Spektrum compatible.

I have requested some doc updates from Spektrum describing all this, current implementation is based on mail conversations and T&E.

Reply:
@AndersHoglund: We could add it as a lookup `spektrum_rssi_type` that defaults to `auto`, but allows the user to force a mode if their hardware isn't recognised properly.